controllers/service.go

```diff
@@ -318,18 +341,58 @@ func podServiceNeeded(multiPodPerHost bool, networkPolicy *asdbv1.AerospikeNetwo
 	return networkSet.Len() > 2
 }
 
-func (r *SingleClusterReconciler) createPodServiceIfNeeded(pods []*corev1.Pod) error {
-	if !podServiceNeeded(r.aeroCluster.Status.PodSpec.MultiPodPerHost, &r.aeroCluster.Status.AerospikeNetworkPolicy) &&
+func (r *SingleClusterReconciler) createOrUpdatePodServiceIfNeeded(pods []*corev1.Pod) error {
+	if (!podServiceNeeded(r.aeroCluster.Status.PodSpec.MultiPodPerHost, &r.aeroCluster.Status.AerospikeNetworkPolicy) ||
+		isServiceTLSChanged(r.aeroCluster.Spec.AerospikeConfig, r.aeroCluster.Status.AerospikeConfig)) &&
 		podServiceNeeded(r.aeroCluster.Spec.PodSpec.MultiPodPerHost, &r.aeroCluster.Spec.AerospikeNetworkPolicy) {
 		// Create services for all pods if network policy is changed and rely on nodePort service
 		for idx := range pods {
 			if err := r.createPodService(
 				pods[idx].Name, r.aeroCluster.Namespace,
-			); err != nil && !errors.IsAlreadyExists(err) {
-				return err
+			); err != nil {
+				if !errors.IsAlreadyExists(err) {
+					return err
+				}
+
+				if err := r.updatePodServicePorts(
+					pods[idx].Name, r.aeroCluster.Namespace,
+				); err != nil {
+					return err
+				}
 			}
 		}
 	}
 
 	return nil
 }
+
+func isServiceTLSChanged(specAeroConf, statusAeroConf *asdbv1.AerospikeConfigSpec) bool {
+	specTLSName, _ := asdbv1.GetServiceTLSNameAndPort(specAeroConf)
+	statusTLSName, _ := asdbv1.GetServiceTLSNameAndPort(statusAeroConf)
```

Comment:
We should handle nil `statusAeroConf` before checking TLS name and port and accordingly return true or false